lib/TPP/GPU/GpuToCuda.cpp

```diff
@@ -67,18 +67,34 @@ struct GpuToCuda : public tpp::impl::GpuToCudaBase<GpuToCuda>,
         memref::createExpandStridedMetadataPass());
     pm.addNestedPass<gpu::GPUModuleOp>(arith::createArithExpandOpsPass());
     pm.addNestedPass<gpu::GPUModuleOp>(createLowerAffinePass());
+    pm.addNestedPass<gpu::GPUModuleOp>(createConvertVectorToSCFPass());
     pm.addNestedPass<gpu::GPUModuleOp>(createConvertSCFToCFPass());
 
-    // Create CUDA kernels.
-    pm.addNestedPass<gpu::GPUModuleOp>(createStripDebugInfoPass());
+    pm.addNestedPass<gpu::GPUModuleOp>(createConvertNVGPUToNVVMPass());
     pm.addNestedPass<gpu::GPUModuleOp>(createConvertGpuOpsToNVVMOps());
-    pm.addNestedPass<gpu::GPUModuleOp>(createReconcileUnrealizedCastsPass());
+    pm.addNestedPass<gpu::GPUModuleOp>(createConvertVectorToLLVMPass());
+    pm.addNestedPass<gpu::GPUModuleOp>(createConvertSCFToCFPass());
```

Comment:
Duplicate pass?

Reply:
The first one is necessary to "finalize" vector lowering.
This one doesn't seem needed as of right now but I left it for completeness if any other passes generate scf.

Reply:
I'd remove it if not needed, so we can detect if we ever need and why in the future.